### PR TITLE
add label filter

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
This pr will support require_any logic.

File: scheduler/types.go [hunk 1]
```diff
@@ -1,5 +1,9 @@
 package scheduler
 
+const (
+	labelPool = "labelPool"
```
Comment (cjellick) [inline scheduler/types.go]: Can you move this next to the otehr Pool constants in the scheduler.go?

File: scheduler/scheduler.go [hunk 2]
```diff
@@ -52,7 +52,9 @@ func (s *Scheduler) PrioritizeCandidates(resourceRequests []ResourceRequest) ([]
 	sort.Sort(hs)
 	sortedIDs := ids(hs.hosts)
 	filteredHosts := s.PortFilter(resourceRequests, sortedIDs)
+	filteredHosts = s.LabelFilter(filteredHosts, context)
 	s.reserveTempPool(sortedIDs[0], resourceRequests)
+	logrus.Infof("return hosts %v", filteredHosts)
```
Comment (cjellick) [inline scheduler/scheduler.go]: We already log the hosts we are returning when we log the entire event response. Can you remove this log line? I want to a little more protective about over-logging.

File: scheduler/scheduler_label_test.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	c.Assert(actual, check.DeepEquals, []string{"2"})
+
+	actual, err = scheduler.PrioritizeCandidates(nil, context4)
+	c.Assert(err, check.IsNil)
+	c.Assert(actual, check.DeepEquals, []string{"2"})
+
+	actual, err = scheduler.PrioritizeCandidates(nil, context5)
+	c.Assert(err, check.IsNil)
+	c.Assert(actual, check.DeepEquals, []string{})
+}
```
Comment (cjellick) [inline scheduler/scheduler_label_test.go]: Can you add a couple test scenarios for sending more than one context is a request? Meaning the contexdt would looklike []Context{conA, conB}

**Comment by cjellick** [comment]
@StrongMonkey please open an issue for 1.5 for documenting this feature

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/7795

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/7254

File: scheduler/label_filter.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+				pass = false
+				break
+			}
+		}
+		if pass {
+			qualify = true
+			break
+		}
+	}
+	if qualify {
```
Comment (cjellick) [inline scheduler/label_filter.go]: you could just return qulify

**Comment by StrongMonkey** [comment]
@cjellick this pr is also ready to be reviewed now.

**Review by cjellick — CHANGES_REQUESTED** [review]

**Comment by cjellick** [comment]
LGTM

**Review by cjellick — APPROVED** [review]

**Pull request merged**